### PR TITLE
Fix Git error during release workflow

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

First failed run here -> https://github.com/holochain/holochain/actions/runs/9183427539

### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: .github/workflows/holochain-build-and-test.yml [hunk 1]
```diff
@@ -105,16 +105,22 @@ jobs:
...
+          # If a nix store path was passed, ensure it exists locally. This will fetch the path from cachix.
           if [[ ${{ inputs.repo_path }} == "/nix/"* ]]; then
             nix-store --realise ${{inputs.repo_path }}
+            
+            # Temporary workaround for https://github.com/NixOS/nix/issues/10202
+            git config --global --add safe.directory ${{ inputs.repo_path }}
           fi
 
           # first build all derivations. this could be enough for test derivations.
           nix build -L --show-trace \
```
Comment (ThetaSinner) [inline .github/workflows/holochain-build-and-test.yml:117]: Moving to Nix 2.20 here -> https://github.com/holochain/holochain-infra/pull/92

Has caused an issue with Git (libgit2 embedded in Nix) rejecting files owned by `root` as in input -> https://github.com/holochain/holochain/actions/runs/9191914450/job/25280481018#step:4:90

File: .github/workflows/holochain-build-and-test.yml [hunk 2]
```diff
@@ -105,16 +105,22 @@ jobs:
 
           git status
 
-          # if a nix store path was passed, ensure it exists locally
+          # If a nix store path was passed, ensure it exists locally. This will fetch the path from cachix.
           if [[ ${{ inputs.repo_path }} == "/nix/"* ]]; then
             nix-store --realise ${{inputs.repo_path }}
+            
+            # Temporary workaround for https://github.com/NixOS/nix/issues/10202
+            git config --global --add safe.directory ${{ inputs.repo_path }}
```
Comment (ThetaSinner) [inline .github/workflows/holochain-build-and-test.yml:113]: Asks Git to accept that the current user and the owner of the repo fetched from Cachix are different. See the next comment

File: .github/workflows/release-prepare.yml [hunk 3]
```diff
@@ -209,9 +209,9 @@ jobs:
           export LATEST_HOLOCHAIN_RELEASE_TAG=$(git tag --sort=-taggerdate --merged "${RELEASE_BRANCH}" | grep holochain- | head -n1)
           export LATEST_HOLOCHAIN_RELEASE_VERSION=${LATEST_HOLOCHAIN_RELEASE_TAG/holochain-/}
 
-          # clean the repo before adding it to the store
+          # clean the repo before adding it to the store. Adding the directory to the store will cause the content to be pushed to cachix.
           git clean -ffdx
-          export STORE_PATH=$(nix store add-path --name holochain_repo .)
+          export STORE_PATH=$(nix store add --name holochain_repo .)
```
Comment (ThetaSinner) [inline .github/workflows/release-prepare.yml:214]: Deprecated command, new one does the same thing with the same arguments

Comment (ThetaSinner) [reply .github/workflows/release-prepare.yml:214]: Just thought i'd fix it while I'm here!

**Review by neonphog — APPROVED** [review]

**Pull request merged**